wolfcrypt/src/ed25519.c

```diff
@@ -735,22 +822,43 @@ int wc_ed25519_import_private_only(const byte* priv, word32 privSz,
     return 0;
 }
 
-/*
-    For importing a private key and its associated public key.
+
+/* Import an ed25519 private and public keys from byte array(s).
+ *
+ * priv    [in]  Array holding private key from wc_ed25519_export_private_only(),
+ *               or private+public keys from wc_ed25519_export_private().
+ * privSz  [in]  Number of bytes of data in private key array.
+ * pub     [in]  Array holding public key (or NULL).
+ * pubSz   [in]  Number of bytes of data in public key array (or 0).
+ * key     [in]  Ed25519 private/public key.
+ * returns BAD_FUNC_ARG when a required parameter is NULL or an invalid
+ *         combination of keys/lengths is supplied, 0 otherwise.
  */
 int wc_ed25519_import_private_key(const byte* priv, word32 privSz,
                                 const byte* pub, word32 pubSz, ed25519_key* key)
 {
     int    ret;
 
     /* sanity check on arguments */
-    if (priv == NULL || pub == NULL || key == NULL)
+    if (priv == NULL || key == NULL)
         return BAD_FUNC_ARG;
 
     /* key size check */
-    if (privSz < ED25519_KEY_SIZE || pubSz < ED25519_PUB_KEY_SIZE)
+    if (privSz < ED25519_KEY_SIZE)
```

Comment:
ED25519_KEY_SIZE is private key only.
ED25519_PRV_KEY_SIZE is both public and private.
Therefore private key can't contain public key with this check.

Reply:
I think you're reading the inequality backwards so to speak.  The test assures `privSz` is _at least_ `ED25519_KEY_SIZE`, but allows that it may be bigger, i.e. `ED25519_PRV_KEY_SIZE`.

Reply:
You're right!